server/region_statistics.go

```diff
@@ -107,8 +111,47 @@ func (r *regionStatistics) Observe(region *core.RegionInfo, stores []*core.Store
 	if len(stores) == 0 {
 		return
 	}
+	r.labelLevelStats.Observe(region, stores, labels)
+}
+
+func (r *regionStatistics) Collect() {
+	regionStatusGauge.WithLabelValues("miss_peer_region_count").Set(float64(len(r.stats[missPeer])))
+	regionStatusGauge.WithLabelValues("extra_peer_region_count").Set(float64(len(r.stats[extraPeer])))
+	regionStatusGauge.WithLabelValues("down_peer_region_count").Set(float64(len(r.stats[downPeer])))
+	regionStatusGauge.WithLabelValues("pending_peer_region_count").Set(float64(len(r.stats[pendingPeer])))
+	regionStatusGauge.WithLabelValues("incorrect_namespace_region_count").Set(float64(len(r.stats[incorrectNamespace])))
+	r.labelLevelStats.Collect()
+}
+
+type labelLevelStatistics struct {
+	regionLabelLevelStats map[uint64]int
+	labelLevelCounter     map[int]int
+}
+
+func newLabelLevlStatistics() *labelLevelStatistics {
+	return &labelLevelStatistics{
+		regionLabelLevelStats: make(map[uint64]int),
+		labelLevelCounter:     make(map[int]int),
+	}
+}
+
+func (l *labelLevelStatistics) Observe(region *core.RegionInfo, stores []*core.StoreInfo, labels []string) {
+	regionID := region.GetId()
 	regionLabelLevel := getRegionLabelIsolationLevel(stores, labels)
-	regionLabelLevelHistogram.Observe(float64(regionLabelLevel))
+	if level, ok := l.regionLabelLevelStats[regionID]; ok {
+		if level == regionLabelLevel {
+			return
+		}
+		l.labelLevelCounter[level]--
```

Comment:
what if region was merged, so there is no heartbeat anymore, namely Observe will not be called. Maybe we should add a function which can delete all relative stats of merged region.